### ISSUE TITLE
Sessions and Cookies inside CurrentUser

### ISSUE THREAD

**Issue by xepozz** [yiisoft/user#74]
CurrentUser should not depend on session/cookies or something runtime-dependent

**Comment by samdark** [comment]
Where should it store its data then?

**Comment by xepozz** [comment]
`LoginMiddleware` / `CookieLoginMiddleware` or any others

**Comment by klsoft-web** [comment]
I don't think this issue is relevant to the yiisoft/user package, so it should be closed.
Perhaps the author was misled by the name of the package.
A more appropriate name for the package would be` yiisoft/auth-session`, since it uses sessions to save and retrieve identity information, in a similar way to the [yiisoft/auth-jwt](https://github.com/yiisoft/auth-jwt) package, which uses JWT tokens from the header to retrieve identity information.
Please also note that the Yiisoft\Session\Session class has a [handler](https://github.com/yiisoft/session/blob/master/src/Session.php#L42) that allows you to save and retrieve session information from any source.
Therefore, any abstraction needs to be related to the [yiisoft/auth](https://github.com/yiisoft/auth) package.
